### PR TITLE
ntjoin 1.0.3

### PR DESCRIPTION
**Pull request by lcoombe**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by lcoombe** [comment]
Seems to be failing on macos when trying to install `pysam`:
```
  Building wheel for pysam (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
```
I can reproduce locally, but I'm not sure why this is happening...
If I do a `pip install pysam` in a virtual environment with python 3.8.3, I don't see the error

**Comment by sjackman** [comment]
```console
$ gcc-10 -undefined dynamic_lookup -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk build/temp.macosx-10.15-x86_64-3.8/pysam/libchtslib.o build/temp.macosx-10.15-x86_64-3.8/pysam/htslib_util.o build/temp.macosx-10.15-x86_64-3.8/htslib/kfunc.o build/temp.macosx-10.15-x86_64-3.8/htslib/knetfile.o build/temp.macosx-10.15-x86_64-3.8/htslib/kstring.o build/temp.macosx-10.15-x86_64-3.8/htslib/bcf_sr_sort.o build/temp.macosx-10.15-x86_64-3.8/htslib/bgzf.o build/temp.macosx-10.15-x86_64-3.8/htslib/errmod.o build/temp.macosx-10.15-x86_64-3.8/htslib/faidx.o build/temp.macosx-10.15-x86_64-3.8/htslib/header.o build/temp.macosx-10.15-x86_64-3.8/htslib/hfile.o build/temp.macosx-10.15-x86_64-3.8/htslib/hfile_net.o build/temp.macosx-10.15-x86_64-3.8/htslib/hts.o build/temp.macosx-10.15-x86_64-3.8/htslib/hts_os.o build/temp.macosx-10.15-x86_64-3.8/htslib/md5.o build/temp.macosx-10.15-x86_64-3.8/htslib/multipart.o build/temp.macosx-10.15-x86_64-3.8/htslib/probaln.o build/temp.macosx-10.15-x86_64-3.8/htslib/realn.o build/temp.macosx-10.15-x86_64-3.8/htslib/regidx.o build/temp.macosx-10.15-x86_64-3.8/htslib/region.o build/temp.macosx-10.15-x86_64-3.8/htslib/sam.o build/temp.macosx-10.15-x86_64-3.8/htslib/synced_bcf_reader.o build/temp.macosx-10.15-x86_64-3.8/htslib/vcf_sweep.o build/temp.macosx-10.15-x86_64-3.8/htslib/tbx.o build/temp.macosx-10.15-x86_64-3.8/htslib/textutils.o build/temp.macosx-10.15-x86_64-3.8/htslib/thread_pool.o build/temp.macosx-10.15-x86_64-3.8/htslib/vcf.o build/temp.macosx-10.15-x86_64-3.8/htslib/vcfutils.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_codecs.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_decode.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_encode.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_external.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_index.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_io.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_samtools.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/cram_stats.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/mFILE.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/open_trace_file.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/pooled_alloc.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/rANS_static.o build/temp.macosx-10.15-x86_64-3.8/htslib/cram/string_alloc.o build/temp.macosx-10.15-x86_64-3.8/htslib/hfile_libcurl.o build/temp.macosx-10.15-x86_64-3.8/htslib/hfile_gcs.o build/temp.macosx-10.15-x86_64-3.8/htslib/hfile_s3.o build/temp.macosx-10.15-x86_64-3.8/htslib/hfile_s3_write.o -L/private/tmp/pip-install-ef54cty0/pysam/pysam -L/private/tmp/pip-install-ef54cty0/pysam/pysam -L/private/tmp/pip-install-ef54cty0/pysam -L/private/tmp/pip-install-ef54cty0/pysam/build/lib.macosx-10.15-x86_64-3.8/pysam -Lbuild/lib.macosx-10.15-x86_64-3.8/pysam -L/usr/local/lib -L/usr/local/opt/openssl@1.1/lib -L/usr/local/opt/sqlite/lib -lz -llzma -lbz2 -lz -lcurl -o build/lib.macosx-10.15-x86_64-3.8/pysam/libchtslib.cpython-38-darwin.so -dynamiclib -rpath @loader_path -Wl,-headerpad_max_install_names -Wl,-install_name,@rpath/libchtslib.cpython-38-darwin.so -Wl,-x    gcc-10: error: loader_path: No such file or directory
    gcc-10: error: unrecognized command-line option '-rpath'
    error: command 'gcc-10' failed with exit status 1
```
This bit `-rpath @loader_path` is wrong and ought to be `-Wl,-rpath=@loader_path`. Looks like a bug in `pysam` to me at first glance.

**Comment by sjackman** [comment]
What part of ntJoin requires OpenMP? One option may be to use clang and `libomp` on macOS rather than `gcc`. The formulae `delly`, `phlawd`, and `treepl`. I generally prefer using `gcc` over `libomp` on macOS, but whatever works is fine.

**Comment by lcoombe** [comment]
Aha good catch -- originally `indexlr` required `openmp` but I believe Vlad has it using pthread.  I'll remove those dependencies specific to `openmp`.

For pysam -- I'm going to try to manually downgrade the version installed and see if that fixes things..

File: Formula/ntjoin.rb [hunk 1]
```diff
@@ -24,14 +22,13 @@ class Ntjoin < Formula
...
 
-  fails_with :clang # needs openmp
-
   def install
     system "make", "-C", "src"
     ENV.prepend_path "PATH", libexec/"bin"
     xy = Language::Python.major_minor_version "python3"
     ENV.prepend_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
     inreplace "requirements.txt", "python-igraph", "python-igraph==0.7.1.post6"
+    inreplace "requirements.txt", "pybedtools", "pysam==0.15.4\npybedtools"
```
Comment (sjackman) [inline Formula/ntjoin.rb:31]: Is this hack needed even when using clang on macOS rather than `gcc`?

Comment (lcoombe) [reply Formula/ntjoin.rb]: Yeah good point -- It installs successfully for me on my local mac without that line (after removing the `openmp` things), so removed it in my latest commit

**Comment by lcoombe** [comment]
Thanks for your help, Shaun!

**Comment by sjackman** [comment]
No worries!

**Pull request merged**